src/main/java/aichan/task/Task.java

```diff
@@ -0,0 +1,100 @@
+package aichan.task;
+
+import aichan.AiChanException;
+
+/**
+ * Represents a task.
+ */
+public class Task {
+    private String taskName;
+    private boolean isDone;
+
+    /**
+     * Constructs a task object.
+     * Initializes description and set isDone to false.
+     *
+     * @param taskName Description of the task.
+     */
+    public Task(String taskName) {
+        this.taskName = taskName;
+        this.isDone = false;
+    }
+
+    /**
+     * Converts a line in the file to its task.
+     *
+     * @param str String of one line in a file.
+     * @return The corresponding task.
+     * @throws AiChanException If the line has incorrect format.
+     */
+    public static Task stringToTask(String str) throws AiChanException {
+        char type = str.charAt(0);
+        boolean isMark = (str.charAt(4) == '1');
+        if (str.length() < 8) {
+            throw new AiChanException("incorrect format in the file");
+        }
+        String description = str.substring(8);
+        if (type == 'T') {
+            Task t = new ToDo(description);
+            if (isMark) t.mark();
+            return t;
+        } else if (type == 'D') {
+            String[] arr = description.split(" \\| ");
+            Task t = new Deadline(arr);
+            if (isMark) t.mark();
+            return t;
+        } else if (type == 'E') {
+            String[] arr = description.split(" \\| | - ");
+            Task t = new Event(arr);
+            if (isMark) t.mark();
+            return t;
+        } else {
+            return null;
+        }
+    }

```

Comment:
* I'm not sure I like how cramped the code is. Maybe can add some **line breaks** to improve readabiliity?
* Not using {} for if statements does not follow the coding standard.

```suggestion
        char type = str.charAt(0);
        boolean isMark = (str.charAt(4) == '1');
        
        if (str.length() < 8) {
            throw new AiChanException("incorrect format in the file");
        }
        
        String description = str.substring(8);
        if (type == 'T') {
            Task t = new ToDo(description);
            if (isMark) {
                t.mark();
            }
            return t;
        } else if (type == 'D') {
            String[] arr = description.split(" \\| ");
            Task t = new Deadline(arr);
            if (isMark) {
                t.mark();
            }
            return t;
        } else if (type == 'E') {
            String[] arr = description.split(" \\| | - ");
            Task t = new Event(arr);
            if (isMark) {
                t.mark();
            }
            return t;
        } else {
            return null;
        }
    }

```